app/models/request.rb

```diff
@@ -0,0 +1,31 @@
+# == Schema Information
+#
+# Table name: requests
+#
+#  id         :integer          not null, primary key
+#  student_id :integer
+#  state      :integer          default(0)
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Request < ActiveRecord::Base
+  validates :student_id, presence: true
+
+  has_many :course_requests
```

Comment:
something that might be useful is to also add this `has_many :through` relation on the course too, if someone wanted to query the requests for a given course